chain/pool/src/lib.rs

```diff
@@ -30,46 +33,36 @@ impl TransactionPool {
         let signer_id = signed_transaction.transaction.signer_id.clone();
         let signer_public_key = signed_transaction.transaction.public_key.clone();
         self.transactions
+            .get_mut()
             .entry((signer_id, signer_public_key))
             .or_insert_with(Vec::new)
             .push(signed_transaction);
     }
 
+    /// Returns a draining structure that pulls transactions from the pool in the proper order.
+    /// It has an option to take transactions with the same key as the last one or with the new key.
+    /// When the iterator is dropped, the rest of the transactions remains in the pool.
+    pub fn draining_iterator(&mut self) -> DrainingIterator {
```

Comment:
```suggestion
   fn draining_iterator(&mut self) -> DrainingIterator {
```

Reply:
In the Part 3, I'm only using `draining_iterator` and not `prepare_transactions`